org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/NamespaceManager.java

```diff
@@ -13,44 +13,50 @@
  *******************************************************************************/
 package org.eclipse.winery.repository.backend;
 
-import org.eclipse.winery.common.ids.Namespace;
-
 import java.util.Collection;
+import java.util.Objects;
+
+import org.eclipse.winery.common.ids.Namespace;
 
 public interface NamespaceManager {
 
-    /**
-     * SIDEFFECT: URI is added to list of known namespaces if it did not exist
-     * before
-     */
     default String getPrefix(Namespace namespace) {
+        Objects.requireNonNull(namespace);
+
         String ns = namespace.getDecoded();
         return this.getPrefix(ns);
     }
 
     /**
-     * SIDEFFECT: URI is added to list of known namespaces if it did not exist
-     * before
+     * Returns a prefix for the given namespace. With two different namespaces, to different prefixes are returned.
+     * The returned prefixes are not persistest. Thus, two instances of a NamespaceManager might return different
```

Comment:
Why is that? Wouldn't this result in failing tests at some point in time?

Reply:
f2f discussion: Single instance vs. changes at `Namespaces.properties`. Current decision. Namespace.properties should take the explicitly user-approved namespaces only. Not the implicitly "passt scho" namespaces.